.github/workflows/build-and-test-windows.yml

```diff
@@ -45,8 +51,7 @@ jobs:
       GOMEMLIMIT: 2GiB
     steps:
       - uses: actions/checkout@v4
-      - if: matrix.group == 'receiver-0'
```

Comment:
Why remove the conditional? Is it needed for something other than `receiver-0` group?

Reply:
I believe it is needed for the iis receiver, which changed to group 1, showing how the condition is fragile.  I removed it so that in the future if the groups change again we wont have to worry about updating this condition.  That does mean that all the groups get IIS installed.  Since the tests don't run that often I thought that trade off was worth it.

Reply:
It adds something like 50s to 1m30s to each Windows unit test run I tend to see as waste, but, overall I think you are right: this extra time and compute seems acceptable in exchange for the consistency.